src/os/bluestore/BlueStore.h

```diff
@@ -1545,21 +1539,24 @@ class BlueStore : public ObjectStore,
     std::atomic<uint64_t> num_extents = {0};
     std::atomic<uint64_t> num_blobs = {0};
     uint64_t buffer_bytes = 0;
-
+    Writings& writings;
```

Comment:
My implementation is arguably too specific to the problem that why I didn't introduce many abstractions. In this case writings is sent to the Buffer Cache but from looking at the code we understand it's the same as the global `writings` stored in BlueStore. I feel like we are over "modularizing" before even needing it.

Reply:
Not sure I understand what's your concern here. Having additional reference in BufferCacheShard? Or having Writings class at all? 
If the former - that's just for the sake of convenience, to avoid passing both shard and writing refs in BufferSpace methods and hence avoid their signatures bloating.

Reply:
My concern probably isn't productive anymore, rather it's a personal question I have, whether I think it's worth a single case implementation worth a new abstraction as usually I think of abstractions on repeated work.

Reply:
Additionally having modular structure helps in code testing, modification, supportability, readability etc... ;)